src/model/wallet.ts

```diff
@@ -365,10 +367,43 @@ export const loadWalletViem = async <S extends Screen>({
   const selectedWallet = getWalletWithAccount(addressToUse);
   const isHardwareWallet = selectedWallet?.type === walletTypes.bluetooth;
 
-  // For now, we'll skip hardware wallet support in viem version
-  // Hardware wallets would need special handling with viem
+  // Handle Ledger hardware wallets
   if (isHardwareWallet) {
-    logger.debug('[wallet]: Hardware wallets not yet supported in viem version');
+    // Attempt to load the hardware key ("deviceId/index") stored for this address
+    const hardwareKey = await loadPrivateKey(addressToUse, true);
+
+    // Bail out if user cancelled or isn't authenticated
+    if (hardwareKey === kc.ErrorType.UserCanceled || hardwareKey === kc.ErrorType.NotAuthenticated) {
+      return null;
+    }
+
+    if (hardwareKey && typeof hardwareKey === 'string' && hardwareKey.includes('/')) {
+      const [deviceId, indexStr] = hardwareKey.split('/');
+      const derivationIndex = Number(indexStr) || 0;
+
+      try {
+        const chainForLedger = publicClient.chain ?? mainnet;
+        const chainId = chainForLedger.id || ChainId.mainnet;
```

Comment:
fallback never used

might be better to use `??` in case chainId can be 0, as planned in some intents afaik